Mathlib/Logic/Basic.lean

```diff
@@ -951,6 +951,9 @@ either branch to `a`. -/
 theorem ite_apply (f g : ∀ a, σ a) (a : α) : (ite P f g) a = ite P (f a) (g a) :=
   dite_apply P (fun _ ↦ f) (fun _ ↦ g) a
 
+section
+variable [Decidable Q]
```

Comment:
This one looks wrong, you're just adding an assumption.

Reply:
Previously we were using an explicit `Decidable P`, and the classical decidable instance for `Q`. Surely this is not what was intended?

Reply:
Thanks for the explanation, I had not noticed the `attribute [local instance 10] Classical.propDecidable` at the beginning of the file (which we should probably remove, but that's for another PR).